units/sdlhaptic.inc

```diff
@@ -739,417 +767,541 @@ type
   PSDL_HapticEffect = ^TSDL_HapticEffect;
   TSDL_HapticEffect = record
     {* Common for all force feedback effects *}
-    _type: cuint16;                  {**< Effect type. *}
+    type_: cuint16;                           {**< Effect type. *}
```

Comment:
I'm quite sure this is wrong (which means it's been wrong for quite some time), and will create the equivalent of:
```c
struct TSDL_HapticEffect {
    type: Uint16;
    union {
        ...
    }
}
```
The correct way to do this is:
```
TSDL_HapticEffect = record
    case cuint16 of
        0: (type_: cuint16;);
        SDL_HAPTIC_CONSTANT: (constant: TSDL_HapticConstant;);
        ...
end;
```
This is the way we've done it for `TSDL_Event` and that works fine.

Reply:
You are completly right! - This way you can request the `type_` field for any variant member. I changed the tag integer to `cint`.

Reply:
Fixed in 56f3aabc58180a9130439b5b923c98b0b3adeeb9.